src/test/java/org/cactoos/io/ReaderOfTest.java

```diff
@@ -40,18 +41,317 @@
  * @checkstyle JavadocMethodCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.AvoidDuplicateLiterals"})
 public final class ReaderOfTest {
 
     @Test
-    public void readsSimpleFileContent() throws IOException {
-        final Path temp = Files.createTempFile("cactoos-1", "txt-1");
+    public void readsCharVarArg() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(new ReaderOf('a', 'b', 'c')),
+            new InputHasContent("abc")
```

Comment:
@proshin-roman Is it not possible to use Hamcrest matchers here? It seems strange to me, to use matchers from cactoos itself -- what if the matcher will have a bug in the future? Tests will break for that reason and it will be misleading. 

I would use trusted Hamcrest matchers, even if you have to call ``asString()`` or something, on ``InputOf``

Furthermore, it would be nice if we could test ``ReaderOf`` without having to wrap it inside ``InputOf`` -- what if the test fails because there is a bug in ``InputOf`` and ``ReaderOf`` is actually fine?

Reply:
I've tried to do that using one private method that does the assertion.